cpp_exercise/tests/linuxipc_test.cc

```diff
@@ -0,0 +1,183 @@
+#include "src/lib/linuxipclib.h"
+#include "src/lib/filehandlerlib.h"
+#include "src/lib/ipcpipelib.h"
+
+#include <map>
+#include <vector>
+#include<thread>
+
+#include "gtest/gtest.h"
+
+
+
+class FileHandlerTests : public ::testing::Test{
+
+      public:
+
+      protected:
+      FileHandler nonexFile{"thisfiledoesnotexist.txt"};
+      FileHandler txtFile{"testfile.txt"};
+      FileHandler pdfFile{"pdftestfile.pdf"};
+      std::string samplestring = "Test Data";
+      std::vector<unsigned char> samplevec;
+      
+    
+ };
+
+
+
+  TEST_F(FileHandlerTests, ReadingFromNoExistingFile){
+
+    remove("thisfiledoesnotexist.txt");
+    std::string exception;
+
+     try{
```

Comment:
you can use 'EXCEPT_THROW' macro

Reply:
I tried to used it but since I used customized exception in ipcexceptionlib it did not work

Reply:
really? this macro should work with custom exceptions

Reply:
It was my bad. In my last commit I used ASSERT_THROW and it works